### PR TITLE
Prune_me

### PR DESCRIPTION
**Pull request by okhlif**
### Description
_________________________________________________________________________________

- Reduce ME search region based on HME distortion
- Reduce ME search region based on HME distortion and HME MV
- Adapt half Pel search mode based on fulpel distortion

### Author(s)
_________________________________________________________________________________
@chkngit 
@NaderMahdi 
@okhlif 

### Type of change
_________________________________________________________________________________
Enhancement.
### Tests and performance
_________________________________________________________________________________
Data generated on the full objective-1-fast data set in context in enc-mode 0.

Speed Deviation | BD-Rate Deviation
-- | --
tbd% | tbd%

### CODE REVIEW

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 1]
```diff
@@ -45,9 +45,9 @@ extern "C" {
 
 #define MUS_ME_FP            1 //MUlti-Stage ME - Decouple full-pel from subpel
 #define MUS_ME_FP_SB         1 //MUlti-Stage ME - Full-pel pruning
-#define SKIP_ME_BASED_ON_HME     0 //MUlti-Stage ME - Reduce SR based on HME distortion
-#define REDUCE_ME_FOR_LOW_M_SB   0 //MUlti-Stage ME - Reduce SR based on HME distortion and HME MV
-#define SWITCHED_HALF_PEL_MODE   0 //MUlti-Stage ME - adapt half_pel mode based on fulpel distortion
+#define SKIP_ME_BASED_ON_HME     1 //MUlti-Stage ME - Reduce SR based on HME distortion
+#define REDUCE_ME_FOR_LOW_M_SB   1 //MUlti-Stage ME - Reduce SR based on HME distortion and HME MV
+#define SWITCHED_HALF_PEL_MODE   1 //MUlti-Stage ME - adapt half_pel mode based on fulpel distortion
```
Comment (fouedb) [inline Source/Lib/Common/Codec/EbDefinitions.h]: //Multi-Stage ME - Adapt half_pel mode based on fullpel distortion

File: Source/Lib/Common/Codec/EbDefinitions.h [hunk 2]
```diff
@@ -45,9 +45,9 @@ extern "C" {
 
 #define MUS_ME_FP            1 //MUlti-Stage ME - Decouple full-pel from subpel
 #define MUS_ME_FP_SB         1 //MUlti-Stage ME - Full-pel pruning
-#define SKIP_ME_BASED_ON_HME     0 //MUlti-Stage ME - Reduce SR based on HME distortion
-#define REDUCE_ME_FOR_LOW_M_SB   0 //MUlti-Stage ME - Reduce SR based on HME distortion and HME MV
-#define SWITCHED_HALF_PEL_MODE   0 //MUlti-Stage ME - adapt half_pel mode based on fulpel distortion
+#define SKIP_ME_BASED_ON_HME     1 //MUlti-Stage ME - Reduce SR based on HME distortion
+#define REDUCE_ME_FOR_LOW_M_SB   1 //MUlti-Stage ME - Reduce SR based on HME distortion and HME MV
```
Comment (fouedb) [inline Source/Lib/Common/Codec/EbDefinitions.h]: Multi-stage ME

File: Source/Lib/Encoder/Codec/EbMotionEstimationProcess.c [hunk 3]
```diff
@@ -254,14 +254,13 @@ EbErrorType signal_derivation_me_kernel_oq(SequenceControlSet *       scs_ptr,
         context_ptr->me_context_ptr->use_subpel_flag = scs_ptr->static_config.enable_subpel;
     if (MR_MODE) {
         context_ptr->me_context_ptr->half_pel_mode    = EX_HP_MODE;
-    } else if (enc_mode == ENC_M0) {
+    } else if (enc_mode <= ENC_M0) {
```
Comment (bzhutencent) [inline Source/Lib/Encoder/Codec/EbMotionEstimationProcess.c]: Is it possible that enc_mode < ENC_M0?

Comment (okhlif) [reply Source/Lib/Encoder/Codec/EbMotionEstimationProcess.c]: No , we usually do switch settings between enc mode with <=
If we ever adopt enc_mode < ENC_M0 , everything works.

**Pull request merged**